node/cn/api_backend.go

```diff
@@ -91,11 +91,35 @@ func (b *CNAPIBackend) HeaderByNumber(ctx context.Context, blockNr rpc.BlockNumb
 	}
 	header := b.cn.blockchain.GetHeaderByNumber(uint64(blockNr))
 	if header == nil {
-		return nil, fmt.Errorf("the block does not exist (block number: %d)", blockNr)
+		return nil, fmt.Errorf("the header does not exist (block number: %d)", blockNr)
 	}
 	return header, nil
 }
 
+func (b *CNAPIBackend) HeaderByNumberOrHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (*types.Header, error) {
+	if blockNr, ok := blockNrOrHash.Number(); ok {
+		return b.HeaderByNumber(ctx, blockNr)
+	}
+	if hash, ok := blockNrOrHash.Hash(); ok {
+		header, err := b.HeaderByHash(ctx, hash)
+		if err != nil {
+			return nil, err
+		}
+		if header == nil {
```

Comment:
Is it possible for the `headr` to be `nil`?

Reply:
Oh.. header never be nil.
Thanks jk!